### PR TITLE
Deterministic Subsetting LB policy

### PR DESCRIPTION
**Pull request by s-matyukevich**
A POC for https://github.com/grpc/proposal/pull/383

### CODE REVIEW

File: vet.sh [hunk 1]
```diff
@@ -82,7 +82,7 @@ not git grep -l 'x/net/context' -- "*.go"
 
 # - Do not import math/rand for real library code.  Use internal/grpcrand for
 #   thread safety.
-git grep -l '"math/rand"' -- "*.go" 2>&1 | not grep -v '^examples\|^stress\|grpcrand\|^benchmark\|wrr_test'
+git grep -l '"math/rand"' -- "*.go" 2>&1 | not grep -v '^examples\|^stress\|grpcrand\|^benchmark\|wrr_test\|deterministicsubsetting'
```
Comment (s-matyukevich) [inline vet.sh:85]: We need to use math/rand directly because the algorithm uses round index as seed, so that shuffle results are deterministic.

**Comment by easwars** [comment]
@s-matyukevich : Thank you for this PR and the associated gRFC. One of our TLs is away for a couple of weeks. We will start reviewing the gRFC once they are back. Please expect some delays here. Thanks.

**Comment by easwars** [comment]
Blocked on the gRFC: https://github.com/grpc/proposal/pull/383

**Comment by dfawley** [comment]
@s-matyukevich thank you for the PR.  I'm going to close this for now, pending [the gRFC](https://github.com/grpc/proposal/pull/383)

**Pull request closed without merging**